### PR TITLE
gazelle: use default proto generation except for testgrid/config

### PR DESCRIPTION
**Pull request by ixdy**
We should also fix `testgrid/config` at some point, but that's slightly more involved, as I noted in https://kubernetes.slack.com/archives/C09QZ4DQB/p1513824406000116.

/assign @BenTheElder @fejta

### CODE REVIEW

**Comment by ixdy** [comment]
This collides with https://github.com/kubernetes/test-infra/pull/6149 and I'm not sure how to address that.

/shrug

**Comment by ixdy** [comment]
Actually, running `hack/prune-libraries.sh --fix` mostly did the right thing, and ended up in a net reduction of lines in `vendor/`. It seems confused about `vendor/github.com/prometheus/client_model/go` though.

**Comment by ixdy** [comment]
`hack/prune-dependencies.sh` keeps choking on `vendor/github.com/prometheus/client_model/BUILD`. I think maybe it's trying to delete it twice?

**Comment by ixdy** [comment]
I think I figured it out - we needed one more call to `hack/update-bazel.sh` after pruning to get things into a steady state.

File: vendor/BUILD [hunk 1]
```diff
@@ -43,7 +43,6 @@ filegroup(
         "//vendor/github.com/matttproud/golang_protobuf_extensions/pbutil:all-srcs",
         "//vendor/github.com/peterbourgon/diskv:all-srcs",
         "//vendor/github.com/prometheus/client_golang/prometheus:all-srcs",
-        "//vendor/github.com/prometheus/client_model:all-srcs",
```
Comment (kargakis) [inline vendor/BUILD]: Shouldn't this also update the dep files?

Comment (kargakis) [reply vendor/BUILD]: https://github.com/kubernetes/test-infra/blob/cab53df75a87c008fafb0a6036652110905656c4/Gopkg.lock#L187

Comment (ixdy) [reply vendor/BUILD]: yeah, it should have.

Comment (kargakis) [reply vendor/BUILD]: Also, this library seems to be used transitively by `github.com/prometheus/client_golang`.

Comment (kargakis) [reply vendor/BUILD]: OK, this seems to be added back in this very same PR. Sorry for the noise.

**Pull request merged**